ui/src/context/terminalContext.ts

```diff
@@ -92,13 +88,43 @@ export const createTerminalContext = (): TerminalContext => {
     });
 
     const handLunaCommand = (msg: LunaMessage) => {
+      const terminalStore = useTerminalStore();
+      const currentTab = terminalStore.currentTab;
+
+      // 只有在 k8s 连接或切换的时候 currentTab 才会有值
+      if (currentTab) {
+        const treeStore = useTreeStore();
+        const currentNode = treeStore.getTerminalByK8sId(currentTab);
+
+        if (!currentNode || !currentNode.terminal) {
+          console.warn('No active K8s terminal instance found');
+          return;
+        }
+
+        try {
+          currentNode.socket.send(
+            JSON.stringify({
+              id: currentNode.id,
+              k8s_id: currentNode.k8s_id,
+              type: 'TERMINAL_K8S_DATA',
+              data: msg.data,
+            })
+          );
+        } catch (error) {
+          console.error('Failed to paste command to K8s terminal:', error);
+        }
+
+        return;
+      }
+
       const socket = connectionStore.socket;
       const terminalId = connectionStore.terminalId;
 
       if (!socket || !terminalId) {
         console.error('WebSocket connection may be closed, please refresh the page');
         return;
       }
+
       socket.send(formatMessage(terminalId, FORMATTER_MESSAGE_TYPE.TERMINAL_DATA, msg.data));
     };
 
```

Comment:
This does not look like it has any known differences that would warrant an investigation, but I would perform a detailed review of each method before proceeding with integration. The main concerns seem to be around using global types, such as `any`, which are usually avoided for performance and maintainability benefits when defining events for multiple instances.

However in TypeScript context where `@ts-expect-error` can be used, you need to ensure the type is properly annotated instead of relying on `any`. Here is something more clear:

```
type TEventType = keyof typeof TerminalEvents;

interface EventContext extends NodeGlobalTS {
  // etc...
}
```